File: src/main/java/org/voegtle/weatherstation/server/ReportingServlet.java
package org.voegtle.weatherstation.server;

import org.voegtle.weatherstation.server.logic.AdminNotifier;
import org.voegtle.weatherstation.server.request.ResponseCode;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

public class ReportingServlet extends AbstractServlet {

  protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
    AdminNotifier notifier = new AdminNotifier(pm, locationProperties);
    notifier.notifiy();
    returnResult(response, ResponseCode.ACKNOWLEDGE);
  }
}

File: src/main/java/org/voegtle/weatherstation/server/ReportingServlet.kt
package org.voegtle.weatherstation.server

import jakarta.servlet.ServletException
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.voegtle.weatherstation.server.logic.AdminNotifier
import org.voegtle.weatherstation.server.request.ResponseCode
import java.io.IOException

class ReportingServlet : AbstractServlet() {

  @Throws(ServletException::class, IOException::class)
  override fun doGet(request: HttpServletRequest, response: HttpServletResponse) {
    val notifier = AdminNotifier(pm, locationProperties!!)
    notifier.sendNotifications()
    returnResult(response, ResponseCode.ACKNOWLEDGE)
  }
}
